docs/iris/src/userguide/real_and_lazy_data.rst

```diff
@@ -1,246 +1,211 @@
-    >>> cube.has_lazy_data()
-    False
 
-You can also convert realized data back into a lazy array:
+This is referred to as **lazy evaluation**.
 
-.. doctest::
+Certain operations, including regridding and plotting, can only be run on real data.
+Calling such operations on lazy data will automatically realise your lazy data.
 
-    >>> cube.has_lazy_data()
-    False
-    >>> cube.data = cube.lazy_data()
-    >>> cube.has_lazy_data()
-    True
+You can also realise (and so load into memory) your cube's lazy if you 'touch' the data.
+This means directly accessing the data by calling ``cube.data``, as in the previous example.
 
-Core data refers to the current state of the cube's data, be it real or
-lazy.  This can be used if you wish to refer to the data array but are
-indifferent to its current state.  If the cube's data is lazy, it will not be
-realized when you reference the core data attribute (?):
+Core data
+^^^^^^^^^
 
-.. doctest::
+Cubes have the concept of "core data". This returns the cube's data in its
+current state. If a cube has lazy data, calling the cube's
+:meth:`~iris.cube.Cube.core_data` method will return the cube's lazy dask array.
+If the cube has real data, calling the cube's
+:meth:`~iris.cube.Cube.core_data` method will return the cube's real NumPy array.
+
+Calling the cube's :meth:`~iris.cube.Cube.core_data` method will not change the
+state of the cube's data.  Thus, if the cube's data is lazy then calling
+:meth:`~iris.cube.Cube.core_data` will return the cube's lazy data and not
+realise it.
+
+For example::
 
     >>> cube = iris.load_cube(iris.sample_data_path('air_temp.pp'))
     >>> cube.has_lazy_data()
     True
     >>> the_data = cube.core_data()
+    >>> type(the_data)
+    <class 'dask.array.core.Array'>
     >>> cube.has_lazy_data()
     True
-    >>> real_data = cube.data
+    >>> cube.data
+    >>> the_data = cube.core_data()
+    >>> type(the_data)
+    <type 'numpy.ndarray'>
     >>> cube.has_lazy_data()
     False
 
 
-Changing a Cube's Data
-----------------------
+Coordinates
+-----------
 
-There are several methods of modifying a cube's data array, each one subtly
-different from the others.
+In the same way that Iris cubes contain a data array, Iris coordinates contain
+points and bounds arrays. Coordinate points and bounds arrays can also be real or lazy:
 
-Maths
-^^^^^
+ * A :class:`~iris.coords.DimCoord` will only ever have **real** points and bounds
+   arrays because of monotonicity checks that realise lazy arrays. 
+ * An :class:`~iris.coords.AuxCoord` can have **real or lazy** points and bounds.
+ * An :class:`~iris.aux_factory.AuxCoordFactory` (or derived coordinate) 
+   can have **real or lazy** points and bounds. If all of the 
+   :class:`~iris.coords.AuxCoord` instances that the coordinate is derived from have
+   real points and bounds then the derived coordinate will also have real points
+   and bounds, otherwise the derived coordinate will have lazy points and bounds.
```

Comment:
Is this correct?  So if some of my aux coords are real and some are lazy, the derived coord will be lazy? (By 'be lazy' I mean of course 'have lazy points and bounds', I'm just being lazy, by which I mean 'seeking to minimize my workload')